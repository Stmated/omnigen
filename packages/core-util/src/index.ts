
export * from './equality/index.js';
export * from './interpret/index.js';
export * from './options/index.js';
export * from './parse/index.js';
export * from './util/index.js';
export * from './visit/index.js';
export * from './write/index.js';
