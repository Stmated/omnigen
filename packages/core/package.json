{
  "name": "@omnigen/core",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@omnigen/core-log": "workspace:*",
    "@etclabscore/string-json-template-language": "^1.0.0",
    "@open-rpc/schema-utils-js": "^1.16.1",
    "change-case": "^4.1.2",
    "deepmerge": "^4.3.0",
    "js-yaml": "^4.1.0",
    "json-pointer": "^0.6.2",
    "node-fetch": "^2.6.9",
    "object-hash": "^3.0.0",
    "string-similarity": "^4.0.4",
    "ts-json-schema-generator": "^1.2.0",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@swc/core": "^1.3.32",
    "@types/json-pointer": "^1.0.31",
    "@types/json-schema": "^7.0.15",
    "@types/node": "^18.13.0",
    "@types/node-fetch": "^2.6.2",
    "@types/object-hash": "^2.2.1",
    "@types/string-similarity": "^4.0.0",
    "colors": "^1.4.0",
    "nock": "^13.3.0",
    "typescript": "^5.4.2",
    "type-fest": "^3.6.1"
  }
}
