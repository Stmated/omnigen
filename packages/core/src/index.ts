import {PluginAutoRegistry} from '@omnigen/core-plugin';
import {CommonTransformPlugin, CommonTransform2Plugin, fileWriter, CorePlugin} from './CoreUtilPluginInit.ts';

export * from './ast';
export * from './equality';
export * from './parse';
export * from './util';
export * from './visit';
export * from './write';
export * from './CoreUtilPluginInit.ts';

export default PluginAutoRegistry.register([CorePlugin, CommonTransformPlugin, CommonTransform2Plugin, fileWriter]);
