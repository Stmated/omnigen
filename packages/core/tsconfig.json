{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@omnigen/config-typescript/build.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "baseUrl": "./src"
  }
}
