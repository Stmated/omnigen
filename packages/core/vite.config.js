import {resolve} from 'path';
import {defineConfig} from 'vite';

export default defineConfig({
  build: {
    outDir: 'dist/lib',
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled into your library
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up', 'fs/promises', 'fs'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
      treeshake: 'safest',
    },
  },
  optimizeDeps: {
    exclude: ['sync-fetch'],
  },
});
