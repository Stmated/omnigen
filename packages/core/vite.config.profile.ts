import {resolve} from 'path';
import {type UserConfig} from 'vite';

export default {
  build: {
    outDir: 'dist/profile',
    minify: false,
    terserOptions: {
      compress: false,
      mangle: false,
    },
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.profiling.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
    },
  },
} satisfies UserConfig;
