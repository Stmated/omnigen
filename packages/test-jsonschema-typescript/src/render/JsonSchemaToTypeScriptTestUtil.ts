import {RenderedCompilationUnit, TargetOptions} from '@omnigen/api';
import {PluginManager} from '@omnigen/plugin';
import {TypeScriptOptions, TypeScriptPlugins} from '@omnigen/target-typescript';
import {LoggerFactory} from '@omnigen/core-log';
import {JsonSchemaPlugins} from '@omnigen/parser-jsonschema';
import {FileContext, TargetContext} from '@omnigen/core-plugin';
import {DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS} from '@omnigen/utils-test';

const logger = LoggerFactory.create(import.meta.url);

export class JsonSchemaToTypeScriptTestUtil {

  public static async render(inPath: string, options?: Partial<TypeScriptOptions> & Partial<TargetOptions>): Promise<RenderedCompilationUnit[]> {

    const pm = new PluginManager({includeAuto: true});

    logger.info(`Running plugin manager with ${JsonSchemaPlugins.JsonSchemaPlugin.name} and ${TypeScriptPlugins.TypeScriptPluginInit.name}`);

    const ctx: FileContext & TargetContext = {
      file: inPath,
      target: 'typescript',
      arguments: {
        ...DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS,
        ...options,
        target: 'typescript',
      },
    };

    try {
      const result = await pm.execute({
        ctx: ctx,
        stopAt: TypeScriptPlugins.TypeScriptRendererCtxOut,
      });

      return result.result.ctx.compilationUnits;
    } catch (ex) {
      throw LoggerFactory.formatError(ex);
    }
  }
}
