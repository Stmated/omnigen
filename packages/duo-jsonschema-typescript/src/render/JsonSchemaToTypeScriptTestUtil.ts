import {RenderedCompilationUnit, TargetOptions} from '@omnigen/core';
import {PluginManager} from '@omnigen/plugin';
import {TypeScriptOptions, TypeScriptPlugins} from '@omnigen/target-typescript';
import {LoggerFactory} from '@omnigen/core-log';
import {JsonSchemaPlugins} from '@omnigen/parser-jsonschema';
import {FileContext, TargetContext} from '@omnigen/core-plugin';
import {JavaOptions, SerializationLibrary} from '@omnigen/target-java';

const logger = LoggerFactory.create(import.meta.url);

export class JsonSchemaToTypeScriptTestUtil {

  public static async render(inPath: string, options?: Partial<TypeScriptOptions & TargetOptions>): Promise<RenderedCompilationUnit[]> {

    const pm = new PluginManager({includeAuto: true});

    logger.info(`Running plugin manager with ${JsonSchemaPlugins.JsonSchemaPlugin.name} and ${TypeScriptPlugins.TypeScriptPluginInit.name}`);

    const ctx: FileContext & TargetContext = {
      file: inPath,
      target: 'typescript',
      arguments: {
        ...options,
        ...{
          serializationLibrary: SerializationLibrary.POJO,
        } satisfies Partial<JavaOptions>,
      },
    };

    const result = await pm.execute({
      ctx: ctx,
      stopAt: TypeScriptPlugins.TypeScriptRendererCtxOut,
    });

    return result.result.ctx.compilationUnits;
  }
}
