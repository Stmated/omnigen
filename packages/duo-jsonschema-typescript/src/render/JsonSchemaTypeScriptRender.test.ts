import {describe, expect, test, vi} from 'vitest';
import {Util} from '@omnigen/core-util';
import {JsonSchemaToTypeScriptTestUtil} from './JsonSchemaToTypeScriptTestUtil.ts';
import {RenderedCompilationUnit} from '@omnigen/core';

describe('jsonschema-typescript-render', () => {

  test('discriminator', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const rendered = await JsonSchemaToTypeScriptTestUtil.render(Util.getPathFromRoot('./packages/duo-jsonschema-java/examples/discriminator.json'), {});
    const fileContents = getFileContents(rendered);

    expect(Object.keys(fileContents).sort()).toMatchSnapshot();
    for (const [fileName, content] of Object.entries(fileContents)) {
      expect(content).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });
});

function getFileContents(cus: RenderedCompilationUnit[]): Record<string, string> {

  const result: Record<string, string> = {};

  const fileContents = Map.groupBy(cus, it => it.fileName);
  for (const file of fileContents) {
    result[file[0]] = file[1][0].content;
  }

  return result;
}
