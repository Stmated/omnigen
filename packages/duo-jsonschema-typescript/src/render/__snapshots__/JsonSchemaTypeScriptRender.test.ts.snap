// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsonschema-typescript-render > discriminator 1`] = `
[
  "ResponseData.ts",
  "ResponseDataWithReject.ts",
  "ResponseDataWithResult.ts",
]
`;

exports[`jsonschema-typescript-render > jsonschema7 1`] = `
[
  "IAdditionalProperties.ts",
  "NonNegativeIntegerDefault0.ts",
  "Schema.ts",
  "SchemaDefinitions.ts",
  "SchemaDependencies.ts",
  "SchemaItems.ts",
  "SchemaNonNegativeIntegerDefault0.ts",
  "SchemaPatternProperties.ts",
  "SchemaProperties.ts",
  "SchemaType.ts",
  "SimpleTypes.ts",
]
`;
